viewing_party/party.py

```diff
@@ -1,23 +1,127 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title and genre and rating:
+        movie_dict["title"] = title   
+        movie_dict["genre"] = genre        
+        movie_dict["rating"] = rating
+    else: 
+        return None    
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for item in user_data["watchlist"]:
+        if item["title"] == title:
+            user_data["watched"].append(item)
+            user_data["watchlist"].remove(item)
+    return user_data
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
+def get_watched_avg_rating(user_data):
+    average_rating = 0.0
+    rating_sum = 0.0
+    
+    if len(user_data["watched"]) > 0:
+        for movie in range(len(user_data["watched"])):
+            rating_sum += user_data["watched"][movie]["rating"]  
+        average_rating = rating_sum / len(user_data["watched"])
+    return average_rating 
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    if len(user_data["watched"]) == 0:
+        return None
+    else:

```

Comment:
This `else:` isn't needed.  The if block at the top is an error check that will end the function, but if the conditional is false, the rest of the function will run.  Nesting the code inside the else with a section that isn't nested at the bottom could imply that there are multiple paths to get to lines 49 & 50, but since there are not, all of the code from 44 down will read better if it's all at the same tab level.